CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ option(LOVR_SYSTEM_OPENXR "Use the system-provided OpenXR" OFF)
 option(LOVR_BUILD_EXE "Build an executable (or an apk on Android)" ON)
 option(LOVR_BUILD_SHARED "Build a shared library (takes precedence over LOVR_BUILD_EXE)" OFF)
 option(LOVR_BUILD_BUNDLE "On macOS, build a .app bundle instead of a raw program" OFF)
+option(LOVR_BUILD_WITH_EXPORTED_INTERNALS "When building as a shared library, make all C symbols visible. This exposes internals, so be aware that any APIs you use from C may change at any time." OFF)
```

Comment:
Couple notes:

- The `LOVR_BUILD_*` options all drastically change the output format (exe, dll, app, etc.).  To me, choosing whether symbols are visible does not feel like it falls into this category.  Maybe it should just be a simpler `LOVR_EXPORT` or `LOVR_PUBLIC` or `LOVR_VISIBLE` option.  Also, it does not export internals, it exports everything.
- "When building as a shared library" -- it will also make symbols when building as an executable, right?  Perhaps that clause should be removed.
- The note about "be careful about using the C API", while accurate, feels out of place to me, I would consider removing.

Reply:
Another thought on naming after reading further: Maybe `LOVR_VISIBILITY` that defaults to `hidden` so it can be directly plugged into `C_VISIBILITY_PRESET`?

Reply:
I like the latter option. Implemented.